java/src/main/java/org/whispersystems/libsignal/ecc/Curve.java

```diff
@@ -34,24 +31,18 @@ public static ECPublicKey decodePoint(byte[] bytes, int offset)
       throw new InvalidKeyException("No key type identifier");
     }
 
-    int type = bytes[offset] & 0xFF;
-
-    switch (type) {
-      case Curve.DJB_TYPE:
-        if (bytes.length - offset < 33) {
-          throw new InvalidKeyException("Bad key length: " + bytes.length);
-        }
-
-        byte[] keyBytes = new byte[32];
-        System.arraycopy(bytes, offset+1, keyBytes, 0, keyBytes.length);
-        return new DjbECPublicKey(keyBytes);
-      default:
-        throw new InvalidKeyException("Bad key type: " + type);
+    if (offset == 0) {
```

Comment:
Why is this logic still in Java land?

Reply:
I guess the honest answer would be laziness ;P

Looking at it it would be easy to do this in Rust so I'll add it there and then update here.

Reply:
Fixed!